pkg/serverless/trace/cold_start_span_creator_test.go

```diff
@@ -123,24 +138,35 @@ func TestColdStartSpanCreatorNotColdStart(t *testing.T) {
 	traceAgent := &ServerlessTraceAgent{
 		ta: agnt,
 	}
-	ec := &executioncontext.ExecutionContext{}
-
+	lambdaSpanChan := make(chan *pb.Span)
+	initDurationChan := make(chan float64)
+	stopChan := make(chan struct{})
+	coldStartSpanId := random.Random.Uint64()
 	coldStartSpanCreator := &ColdStartSpanCreator{
-		executionContext: ec,
-		traceAgent:       traceAgent,
-		createSpan:       &sync.Once{},
+		TraceAgent:            traceAgent,
+		CreateSpan:            &sync.Once{},
+		LambdaSpanChan:        lambdaSpanChan,
+		InitDurationChan:      initDurationChan,
+		ColdStartSpanId:       coldStartSpanId,
+		StopChan:              stopChan,
+		SyncSpanDurationMutex: &sync.Mutex{},
 	}
 
+	coldStartSpanCreator.Run()
+	defer coldStartSpanCreator.Stop()
+
 	lambdaSpan := &pb.Span{
 		Service:  "aws.lambda",
-		Name:     "aws.lambda.cold_start",
+		Name:     "aws.lambda.my-function",
 		Start:    time.Now().Unix(),
 		TraceID:  random.Random.Uint64(),
 		SpanID:   random.Random.Uint64(),
 		ParentID: random.Random.Uint64(),
 		Duration: 500,
 	}
-	coldStartSpanCreator.create(lambdaSpan)
+	lambdaSpanChan <- lambdaSpan
+	// Don't write to initDurationChan, as this is not a cold start
+
 	timeout := time.After(time.Second)
```

Comment:
Btw, I was looking at this code yesterday (I was investigating another flaky test) and found that this time after can go down to a millisecond timeout. I believe I found that the usual time for the select to execute is around 80 microseconds.